src/main/kotlin/com/kotlindiscord/database/migrator/MigrationsManager.kt

```diff
@@ -0,0 +1,126 @@
+package com.kotlindiscord.database.migrator
+
+import org.jetbrains.exposed.dao.LongEntity
+import org.jetbrains.exposed.dao.LongEntityClass
+import org.jetbrains.exposed.dao.id.EntityID
+import org.jetbrains.exposed.dao.id.IdTable
+import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.reflections8.Reflections
+
+object Migrations : IdTable<Long>() {
+    override val id = long("id").entityId()
+    val applied = bool("applied").default(false)
+    override val primaryKey = PrimaryKey(id, name = "PK_MIGRATION_ID")
+}
+
+class Migration(id: EntityID<Long>) : LongEntity(id) {
+    companion object : LongEntityClass<Migration>(Migrations)
+
+    var applied by Migrations.applied
+}
+
+/**
+ * Interfaces with the database to manage and apply migrations. Requires a Database connection.
+ */
+class MigrationsManager {
+
+    /**
+     * Gets the id of the migration which was most recently applied to the database.
+     */
+    private val currentMigrationId: Long?
+        get() {
+            var maxMigrationId: Long? = null
+            transaction {
+                SchemaUtils.create(Migrations)
+                val appliedMigrations = Migration.find { Migrations.applied eq true }
+                if (!appliedMigrations.empty()) {
+                    val maxMigration = appliedMigrations.maxBy { it.id }
+                    if (maxMigration != null) maxMigrationId = maxMigration.id.value
+                }
+            }
+
+            return maxMigrationId
+        }
+
+    /**
+     * Map of the user created migrations against their IDs (the time at which they were created).
+     */
+    private val migrationsMap: Map<Long, AbstractMigration> by lazy {
+        // TODO handle the case where we find a non migration object more elegantly
+        val reflections = Reflections("com.kotlindiscord.database.migrations")
+
+        val classes = reflections.getSubTypesOf(AbstractMigration::class.java)
+
+        @Suppress("MagicNumber")
```

Comment:
For the sake of readability, you can move this suppress to the function definition.

I'd then group the `val` statements together and separate them from the conditional below with a blank line.